lib/potluck.rb

```diff
@@ -0,0 +1,57 @@
+require './lib/dish'
+require 'pry'
+
+class Potluck
+  attr_accessor :date,
+                :dishes
+  def initialize(date)
+    @date = date
+    @dishes = []
+  end
+
+  def add_dish(new_dish)
+    @dishes << new_dish
+  end
+
+  def get_all_from_category(selected_category)
+    array_to_return = []
+    @dishes.each do |dish|
+      if dish.category == selected_category
+        array_to_return << dish
+      else
+        next
```

Comment:
This does the trick! No need for the else, though. Possible to have an if without an else.

Also, I don't find this too gross, though some might:

```
array_to_return << dish if dish.category == selected_category
```

Also, Like using each for this as it strengthens the iteration muscles, but could potentially use Ruby's built in `select` method.